base/pmap.jl

```diff
@@ -15,38 +15,81 @@ Note that `f` must be made available to all worker processes; see
 and Loading Packages <man-parallel-computing-code-availability>`)
 for details.
 """
-function pgenerate(p::WorkerPool, f, c)
-    if length(p) == 0
-        return AsyncGenerator(f, c)
+function pgenerate(p::WorkerPool, f, c; config=DEFAULT_PMAP_ARGS)
+    batch_size = config[:batch_size]
+    on_error = config[:on_error]
+    distributed = config[:distributed]
+
+    if  (distributed == false) ||
+        (length(p) == 0) ||
+        (length(p) == 1 && fetch(p.channel) == myid())
+
+        return AsyncGenerator(f, c; on_error=on_error)
+    end
+
+    if batch_size == :auto
+        batches = batchsplit(c, min_batch_count = length(p) * 3)
+    else
+        batches = batchsplit(c, max_batch_size = batch_size)
```

Comment:
Why not retain `min_batch_count = length(p) * 3` in the case where `max_batch_size` is specified?
i.e. to ensure stuff happens in parallel for small input collections.